src/stretchy.js

```diff
@@ -0,0 +1,274 @@
+};
+
+const svgSpan = function(group, options) {
+    // Create a span containing the SVG background-image
+
+    const label = group.value.label.substr(1);
+    let height = 0;
+    let depth = 0;
+    const classArray = ["stretchy"];
+    let fileName = "";
+
+    if (!document.implementation.hasFeature(
+        "http://www.w3.org/TR/SVG11/feature#BasicStructure", "1.1")) {
+
+        // SVG is not supported.
+        height = (label.substr(0, 1) === "x" ? 0.17 : 0.34);
+        depth  = (label.substr(0, 1) === "x" ? 0.17 : 0);
+
+    } else if (utils.contains(["widehat", "widetilde", "undertilde"], label)) {
+
+        // There are four SVG images available for each function.
+        // Choose a taller image when there are more characters.
+        const numChars = group.value.value.length;
+        if (numChars > 5) {
+            height = 0.34;
+            fileName = (label === "widehat" ? "widehat" : "tilde") + "4";
+        } else {
+            const imgIndex = [1, 1, 2, 2, 3, 3][numChars];
+            if (label === "widehat") {
+                height = [0, 0.16, 0.23, 0.23, 0.28, 0.28][numChars];
+                fileName = "widehat" + imgIndex;
+            } else {
+                height = [0, 0.15, 0.195, 0.195, 0.26, 0.26][numChars];
+                fileName = "tilde" + imgIndex;
+            }
+        }
+
+    } else {
+        const imgData = katexImagesData[label];
+        height = imgData[0];
+        depth = imgData[1];
+        fileName = imgData[2];
+
+        if (label.substr(0, 1) === "x") {
+            classArray.push("x-arrow");
+        } else if (utils.contains(["overbrace", "underbrace"], label)) {
+            classArray.push("horiz-brace");
+        } else if (utils.contains(["overbracket", "underbracket"], label)) {
+            classArray.push(label);
+        }
+    }
+
+    const node = buildCommon.makeSpan(classArray, [], options);
+    node.maxFontSize = 1.0;
+
+    if (utils.contains(["overbracket", "underbracket"], label)) {
+        if (group.value.thickness) {
+            const lineWt = group.value.thickness.value + "em";
+            node.style.borderWidth = (group.value.isOver ? lineWt : 0) + " " +
+                lineWt + " " + (group.value.isOver ? 0 : lineWt) + " " + lineWt;
+        }
+
+        if (group.value.height) {
+            height = group.value.height;
+            if (height > 1.0) {
+                node.maxFontSize = 1.1 * height;
+            }
+        }
+
+        if (height !== 0.3) {
+            node.style.height = height + "em";
+        }
+
+    } else if (!options.color) {
+        node.style.backgroundImage = "url(" + katexImgPath() +
+            fileName + ".svg)";
+        // By writing that image path inline, we eliminate ~40 CSS classes.
+        if (height + depth !== 0.34) {
+            node.style.height = (height + depth) + "em";
+        }
+    } else {
+        // Add color. Use CSS mask, if possible
+        if (maskIsSupported()) {
```

Comment:
@edemaine , I can omit the mask in all cases, but the non-mask solution relies on a XMLHttpRequest, which is slow.

@glebm, I would agree with your suggestion. If I interpret [caniuse](http://caniuse.com/#search=mask) correctly, the upcoming version of Firefox will change from partial `mask` support to full support. That would leave behind only IE and Edge. However, I'm not sure that the KaTeX owners would agree with you and me. If Edge does look at the output from such a server-side installation, it will render a span-wide blob of color instead of the subject matter.

So I thank both of you for the education on server-side behavior. This affects not just `\color` but also `\cancel`, another case where I have different code for IE and Edge.

Regarding `\cancel`, I'm going to use background-image SVGs for server-side rendering, just as I use it now for IE and Edge. The only drawback of the SVGs in most browsers is that the line does not get thicker when giant fonts sizes are employed. A 1 px line looks good in most common font sizes, so that is a good fall-back.

On the subject of `\color`, I need some guidance from the KaTeX owners as to what they will accept. The options are:
1. Do not support `\color`.
2. Support `\color` by a (slow) XMLHttpRequest in all cases.
3.  Use `mask` when it is supported. Use `mask` for server-side installations even
though Edge will not render it properly.
4.  Use `mask` when I can use feature detection to verify that it is supported. Use a XMLHttpRequest  for all server-side installations.